src/Pact/Compile.hs

```diff
@@ -151,13 +151,15 @@ reservedAtom = bareAtom >>= \AtomExp{..} -> case HM.lookup _atomAtom reserveds o
   Just r -> commit >> return r
 
 compile :: ParseEnv -> MkInfo -> Exp Parsed -> Either PactError (Term Name)
-compile pe mi e = let ei = mi <$> e in runCompile pe topLevel (initParseState ei) ei
+compile pe mi e = runCompile pe topLevel (initParseState ei) ei
+ where
+  ei = mi <$> e
 
 compileExps :: Traversable t => ParseEnv -> MkInfo -> t (Exp Parsed) -> Either PactError (t (Term Name))
-compileExps pe mi exps = sequence $ compile pe mi <$> exps
+compileExps pe mi exps = mapM (compile pe mi) exps
 
 moduleState :: Compile ModuleState
-moduleState = use (psUser . csModule) >>= \m -> case m of
+moduleState = use (psUser . csModule) >>= \case
```

Comment:
This PR is doing so much more than fixing delta-parsing. Can you submit all of these minor cleanups in their own PR first? That way we can trivially approve and merge those, and then think harder about just the Delta parsing fix.

Reply:
I can, it just is quite a bit of extra work to create new PR each time I fix syntax. There are just so many of those syntactic inconsistencies that make readability of the code unnecessarily hard. Fixing them from time to time along the way is just easier.

Reply:
In this case a LOT of the changes are just fixing syntax, so then it makes sense to bundle all of that up into a PR before making an important change like delta parsing.

Reply:
https://github.com/kadena-io/pact/pull/1197

(The issue is that I don't first read the the whole code base and clean it up and then start working on the feature, but those changes are interleaved with other changes in the PR. For instance, above PR caused merge conflicts during the cherry pick and now I have to wait for it to be squash merged into master before I can merge it back into the original PR, which again will require to resolve the conflict.)